internal/ldso-cache/ldsocache.go

```diff
@@ -307,8 +340,28 @@ func AddLDSOCacheEntriesForDirs(fsys fs.FS, libdirs []string) ([]LDSOCacheEntry,
 	return entries, nil
 }
 
+func LDSOCacheEntriesForDirs(fsys fs.FS, libdirs []string) ([]LDSOCacheEntry, error) {
+	allEntries := []LDSOCacheEntry{}
+
+	for _, libdir := range libdirs {
+		entries, err := ldsoCacheEntriesForDir(fsys, libdir)
+		if err != nil {
+			return nil, err
+		}
+		allEntries = append(allEntries, entries...)
+	}
+
+	// ld expects entries to be reverse-sorted by name. Otherwise
+	// it may report "No such file or directory"
+	sort.Slice(allEntries, func(i, j int) bool {
+		return filepath.Base(allEntries[j].Name) < filepath.Base(allEntries[i].Name)
```

Comment:
is this compatible with non-utf C locale that ld.so expects?

Reply:
Good question. Perhaps I should use text/collate w/ the C locale here.